cmd/spiffe-aws-assume-role/cli/cli_test.go

```diff
@@ -176,6 +194,69 @@ func TestSetsCustomStsEndpoint(t *testing.T) {
 	require.EqualValues(t, stsEndpoint, *_session.Config.Endpoint)
 }
 
+func TestSetsCustomStsRegion(t *testing.T) {
+	stsRegion := uuid.New().String()
+
+	args := []string{
+		"credentials",
+		"--audience=foo",
+		"--role-arn=arn:aws:iam:123456789012:role/foo",
+		fmt.Sprintf("--sts-region=%s", stsRegion),
+		"--spiffe-id=spiffe://foo",
+		"--workload-socket=tcp://127.0.0.1:8080",
+	}
+
+	jwtSource := mocks.JWTSource{}
+	defer jwtSource.AssertExpectations(t)
+	jwtSource.
+		On("FetchToken", mock.Anything).
+		Return("token", nil)
+
+	stsClient := mocks.STSAPI{}
+	defer stsClient.AssertExpectations(t)
+
+	stsCredentials := sts.Credentials{
+		AccessKeyId:     aws.String(""),
+		Expiration:      aws.Time(time.Now()),
+		SecretAccessKey: aws.String(""),
+		SessionToken:    aws.String(""),
+	}
+
+	output := sts.AssumeRoleWithWebIdentityOutput{
+		Credentials: &stsCredentials,
+	}
+
+	stsClient.
+		On("AssumeRoleWithWebIdentityWithContext", mock.Anything, mock.Anything).
+		Return(&output, nil)
+
+	var _session *session.Session
+	stsProvider := func(s *session.Session) stsiface.STSAPI {
+		_session = s
+		return &stsClient
+	}
+
+	context := CliContext{
+		JWTSourceProvider: credentials.StaticJWTSourceProvider(&jwtSource),
+		STSProvider:       stsProvider,
+	}
+
+	failOnError(t, Run(&context, args))
+
+	require.EqualValues(t, stsRegion, *_session.Config.Region)
+}
+
+func TestCreatesSessionWithUnspecifiedEndpointAndRegion(t *testing.T) {
+	require.NotNil(t, createSession("", ""))
+}
+
+func TestCreatesSessionWithCustomEndpointAndRegion(t *testing.T) {
+	endpoint := uuid.New().String()
+	s := createSession(endpoint, "")
+	require.NotNil(t, s)
+	require.EqualValues(t, endpoint, *s.Config.Endpoint)
+}
```

Comment:
Now that Session construction logic has been pulled up a level, these tests are coming with it.